src/vs/base/common/tfIdf.ts

```diff
@@ -75,26 +72,53 @@ export class TfIdfCalculator {
 	 * Count how many times each term (word) appears in a string.
 	 */
 	private static termFrequencies(input: string): TermFrequencies {
-		return countMapFrom(TfIdfCalculator.splitTerms(input));
+		const map = new Map<string, { weight: number; occurrences: number }>();
+		for (const value of TfIdfCalculator.splitTerms(input)) {
+			const existing = map.get(value.term);
+			if (existing) {
+				existing.occurrences++;
+			} else {
+				map.set(value.term, { weight: value.weight, occurrences: 1 });
+			}
+		}
+		return map;
 	}
 
 	/**
 	 * Break a string into terms (words).
+	 *
+	 * TODO: confirm that when we break up a word or generate stems, we likely accidentally over-weight its terms.
+	 * For instance, if the document is: `cats wear hats` and the user searches `cats wear`, we could end up giving too
+	 * much weight to `cats` since the document would be broken into: `[cats, cat, wear, hats, hat]` while the query
+	 * would be broken into `[cats, cat, wear]`. This means that terms derived from `cats` end up being matched on multiple
+	 * times, which isn't really right.
+	 *
+	 * Maybe we need to generate a tree of terms for the document where we stop searching once a match has been found:
 	 */
-	private static *splitTerms(input: string): Iterable<string> {
+	private static *splitTerms(input: string): Iterable<Term> {
 		const normalize = (word: string) => word.toLowerCase();
 
 		// Only match on words that are at least 3 characters long and start with a letter
 		for (const [word] of input.matchAll(/\b\p{Letter}[\p{Letter}\d]{2,}\b/gu)) {
-			yield normalize(word);
+			yield { term: normalize(word), weight: 1 };
 
-			// eslint-disable-next-line local/code-no-look-behind-regex
-			const camelParts = word.split(/(?<=[a-z])(?=[A-Z])/g);
+			// Include both the original term and the stemmed version
+			const stemmedTerm = stem(word);
+			if (stemmedTerm !== word) {
+				yield { term: normalize(stemmedTerm), weight: 0.75 };
+			}
+
+			const camelParts = word.split(/(?=[A-Z])/g);
```

Comment:
Note I changed this Regex due to lookbehind not being available in all browsers. This Regex does still seem to pick up camel cased strings.